arangod/Cluster/HeartbeatThread.cpp

```diff
@@ -748,6 +736,55 @@ void HeartbeatThread::getNewsFromAgencyForCoordinator() {
   }
 }
 
+void HeartbeatThread::handleUserVersionChange(VPackSlice userVersion) {
+  TRI_ASSERT(ServerState::instance()->isCoordinator());
+
+  AuthenticationFeature& af = _server.getFeature<AuthenticationFeature>();
+
+  VPackSlice slice = userVersion[0].get(std::vector<std::string>(
+      {AgencyCommHelper::path(), "Sync", "UserVersion"}));
+
+  if (slice.isInteger()) {
+    // there is a UserVersion
+    uint64_t version = 0;
+    try {
+      version = slice.getUInt();
+    } catch (...) {
+    }
+
+    if (version > 0) {
+      if (af.isActive() && af.userManager() != nullptr) {
+        af.userManager()->setGlobalVersion(version);
+      }
+    }
+  }
+}
+
+void HeartbeatThread::handleFoxxQueueVersionChange(
+    VPackSlice foxxQueueVersion) {
+  TRI_ASSERT(ServerState::instance()->isCoordinator());
+
+  VPackSlice slice = foxxQueueVersion[0].get(std::vector<std::string>(
+      {AgencyCommHelper::path(), "Sync", "FoxxQueueVersion"}));
+
+  if (slice.isInteger()) {
+    // there is a UserVersion
+    uint64_t version = 0;
+    try {
+      version = slice.getUInt();
+    } catch (...) {
+    }
+
+    if (version > 0) {
+      // track the global foxx queues version from the agency. any
+      // coordinator can update this any time. the setQueueVersion
+      // method makes sure we are not going below a value that
+      // we have already seen.
+      _server.getFeature<FoxxFeature>().setQueueVersion(version);
+    }
+  }
```

Comment:
I think in the first few runs it is possible that the agency key `Sync/FoxxQueueVersion` does not yet exist. This case should be handled transparently, i.e. without any warnings/errors, because it is not a problem. I think we can turn the `getUInt()` call into a `getNumber<uint64_t>()` call, which will also handle other numeric types. Still it could run into problems if the stored value is a string or some other datatype. But I would not optimize for these edge cases.